ipn/ipnlocal/local.go

```diff
@@ -1781,6 +1781,7 @@ func (b *LocalBackend) authReconfig() {
 	if err == wgengine.ErrNoChanges {
 		return
 	}
+	b.e.NotifyPrefs(prefs)
```

Comment:
I added an engine callback for prefs updates so that tssocks could be more intelligent about whether or not subnet traffic should go over the proxy dialer.

This works well as I've tested it, but I'm not sure if this is the best place to invoke this callback--or if it's appropriate to expose these prefs to the engine at all.

Reply:
My biggest problem is that the wgcfg.Config.Peers AllowedIPs is different than the netmap.Peers AllowedIPs depending on the value of Prefs.RouteAll.